langchain4j-github-models/src/main/java/dev/langchain4j/model/github/GitHubModelsChatModel.java

```diff
@@ -0,0 +1,425 @@
+package dev.langchain4j.model.github;
+
+import com.azure.ai.inference.ChatCompletionsClient;
+import com.azure.ai.inference.models.ChatCompletions;
+import com.azure.ai.inference.models.ChatCompletionsOptions;
+import com.azure.ai.inference.models.ChatCompletionsResponseFormat;
+import com.azure.core.exception.HttpResponseException;
+import com.azure.core.http.ProxyOptions;
+import dev.langchain4j.agent.tool.ToolSpecification;
+import dev.langchain4j.data.message.AiMessage;
+import dev.langchain4j.data.message.ChatMessage;
+import dev.langchain4j.model.chat.ChatLanguageModel;
+import dev.langchain4j.model.chat.listener.*;
+import dev.langchain4j.model.github.spi.GitHubModelsChatModelBuilderFactory;
+import dev.langchain4j.model.output.FinishReason;
+import dev.langchain4j.model.output.Response;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.time.Duration;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Map;
+import java.util.concurrent.ConcurrentHashMap;
+
+import static dev.langchain4j.data.message.AiMessage.aiMessage;
+import static dev.langchain4j.internal.Utils.copyIfNotNull;
+import static dev.langchain4j.internal.Utils.isNullOrEmpty;
+import static dev.langchain4j.internal.ValidationUtils.ensureNotBlank;
+import static dev.langchain4j.model.github.InternalGitHubModelHelper.*;
+import static dev.langchain4j.spi.ServiceHelper.loadFactories;
+import static java.util.Collections.emptyList;
+import static java.util.Collections.singletonList;
+
+/**
+ * Represents a language model, hosted on GitHub Models, that has a chat completion interface, such as gpt-4o.
+ * <p>
+ * Mandatory parameters for initialization are: gitHubToken (the GitHub Token used for authentication) and modelName (the name of the model to use).
+ * You can also provide your own ChatCompletionsClient instance, if you need more flexibility.
+ * <p>
+ * The list of models, as well as the documentation and a playground to test them, can be found at https://github.com/marketplace/models
+ */
+public class GitHubModelsChatModel implements ChatLanguageModel {
+
+    private static final Logger logger = LoggerFactory.getLogger(GitHubModelsChatModel.class);
+
+    private ChatCompletionsClient client;
+    private final String modelName;
+    private final Integer maxTokens;
+    private final Double temperature;
+    private final Double topP;
+    private final List<String> stop;
+    private final Double presencePenalty;
+    private final Double frequencyPenalty;
+    private final Long seed;
+    private final ChatCompletionsResponseFormat responseFormat;
+    private final List<ChatModelListener> listeners;
+
+    public GitHubModelsChatModel(ChatCompletionsClient client,
```

Comment:
```suggestion
    private GitHubModelsChatModel(ChatCompletionsClient client,
```

Probably makes sense to make them private, otherwise it will be harder to add new parameters without breaking someone's code